### PR TITLE
Add language support

### PR DESCRIPTION
**Pull request by Maufeat**
Adding device language support and added the "Back" string for various languages.

### CODE REVIEW

File: common/language.c [hunk 1]
```diff
@@ -277,6 +275,22 @@ const char* const g_strings[StrId_Max][16] =
         STR_ZH("打开"),
         STR_TW("打开"),
     },
+    
+    [StrId_Actions_Back] =
+    {
+        STR_EN("Back"),
+        STR_ES("Espalda"),
```
Comment (fincs) [inline common/language.c]: lol nope, this means "Back" as in body part! This should be "Atrás". Please do not use google translate; get actual translators to do the job or just leave STR_EN and the others untranslated...

File: common/text.c [hunk 2]
```diff
@@ -1,9 +1,17 @@
 #include "text.h"
 
-//TODO: Update this once libnx supports settings get-language.
-
 static int s_textLang = /*CFG_LANGUAGE_EN*/1;
 
+void textInit(void) {
```
Comment (yellows8) [inline common/text.c]: There's zero error checking here.

Should call setExit at the end of this func too.

File: common/language.c [hunk 3]
```diff
@@ -1,7 +1,5 @@
 #include "language.h"
 
-//TODO: Update this once libnx supports settings get-language.
-
 #define STR_JP(_str) [/*CFG_LANGUAGE_JP*/0] = _str
```
Comment (yellows8) [inline common/language.c]: These defines need updated with the libnx set.h SetLanguage_* enums.

File: common/text.c [hunk 4]
```diff
@@ -1,9 +1,17 @@
 #include "text.h"
 
-//TODO: Update this once libnx supports settings get-language.
-
 static int s_textLang = /*CFG_LANGUAGE_EN*/1;
```
Comment (yellows8) [inline common/text.c]: This should also be updated with a SetLanguage enum.

File: nx_main/main.c [hunk 5]
```diff
@@ -12,11 +12,12 @@ int main(int argc, char **argv)
...
     appletSetScreenShotPermission(1);
-
-    ColorSetId theme;
+                                    
+    ColorSetId theme;   
     setsysInitialize();
     setsysGetColorSetId(&theme);
-    themeStartup((ThemePreset)theme);
+    themeStartup((ThemePreset)theme);    
+    textInit();
```
Comment (plutooo) [inline nx_main/main.c]: Whitespace

File: nx_main/main.c [hunk 6]
```diff
@@ -12,11 +12,12 @@ int main(int argc, char **argv)
     gfxInitDefault();
 
     appletSetScreenShotPermission(1);
-
-    ColorSetId theme;
+                                    
+    ColorSetId theme;
```
Comment (plutooo) [inline nx_main/main.c]: Whitespace

File: nx_main/main.c [hunk 7]
```diff
@@ -12,11 +12,12 @@ int main(int argc, char **argv)
     gfxInitDefault();
 
     appletSetScreenShotPermission(1);
-
-    ColorSetId theme;
+
```
Comment (plutooo) [inline nx_main/main.c]: Whitespace

**Comment by Maufeat** [comment]
Added checks, call `sysExit()` at the end and using now the SetLanguage enum. And ofc the whitespaces. Sorry for those gtranslate strings, I've removed them.

File: common/text.c [hunk 8]
```diff
@@ -1,22 +1,32 @@
...
 void textInit(void) {   
     u64 LanguageCode=0;
     s32 Language=0;
-    setInitialize();
-    setGetSystemLanguage(&LanguageCode);
-    setMakeLanguage(LanguageCode, &Language);
-    s_textLang = Language;
-}
+    Result rc = setInitialize();
+    if (R_FAILED(rc)) s_textLang = SetLanguage_ENUS;
```
Comment (yellows8) [inline common/text.c]: Just init s_textLang here without R_FAILED(rc), and remove the duplicate lines below.

**Comment by yellows8** [comment]
Check for the SWITCH define & use the hard-coded lang-ID values when building for PC.

File: common/text.c [hunk 9]
```diff
@@ -6,16 +6,14 @@ void textInit(void) {
     u64 LanguageCode=0;
     s32 Language=0;
     Result rc = setInitialize();
-    if (R_FAILED(rc)) s_textLang = SetLanguage_ENUS;
+    s_textLang = SetLanguage_ENUS;
     if (R_SUCCEEDED(rc))
```
Comment (yellows8) [inline common/text.c]: The {} aren't needed any more, merge each block into a single line.

**Comment by yellows8** [comment]
SetLanguage_* aren't available for PC.

**Comment by Maufeat** [comment]
Oh ups, I hope the last commit is okay?

File: common/text.c [hunk 10]
```diff
@@ -1,15 +1,36 @@
...
-static int s_textLang = /*CFG_LANGUAGE_EN*/1;
+void textInit(void) {
+    #ifdef SWITCH
+    u64 LanguageCode=0;
+    s32 Language=0;
+    Result rc = setInitialize();
+    s_textLang = SetLanguage_ENUS;
+    if (R_SUCCEEDED(rc)) rc = setGetSystemLanguage(&LanguageCode);
+    if (R_SUCCEEDED(rc)) rc = setMakeLanguage(LanguageCode, &Language);
+    if (R_SUCCEEDED(rc)) s_textLang = Language;
```
Comment (yellows8) [inline common/text.c]: Change it to: ```if (R_SUCCEEDED(rc) && Language < 16) ...```

File: nx_main/main.c [hunk 11]
```diff
@@ -16,7 +16,8 @@ int main(int argc, char **argv)
     ColorSetId theme;
     setsysInitialize();
     setsysGetColorSetId(&theme);
-    themeStartup((ThemePreset)theme);
+    themeStartup((ThemePreset)theme);    
+    textInit();
```
Comment (yellows8) [inline nx_main/main.c]: Missing from pc_main.

**Comment by yellows8** [comment]
Merging but have to disable it since the font doesn't support all used languages.

**Pull request merged**